services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -91,13 +93,15 @@ public Response performLogin(AuthenticationRequest request) {
                 protocolBinding = JBossSAMLURIConstants.SAML_HTTP_POST_BINDING.get();
             }
 
+            String loginHint = request.getHttpRequest().getUri().getQueryParameters().getFirst(LOGIN_HINT_QUERY_PARAMETER);
```

Comment:
This needs to be guarded by a configuration that mimics OIDC identity provider `loginHint` configuration which will be disabled by default:

1. Move `OAuth2IdentityProviderConfig`'s `isLoginHint` and `setLoginHint` to IdentityProviderModel
2. Add a `QueryParam(OIDCLoginProtocol.LOGIN_HINT_PARAM)` to authentication session in `IdentityBrokerService.performLogin` similarly to https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java#L445
3. In this place, only use `request.getAuthenticationSession().getClientNote(OIDCLoginProtocol.LOGIN_HINT_PARAM)`

Also update `realm-identity-provider-saml.html` to configure loginHint similarly to `realm-identity-provider-oidc.html`, and ensure that the label and label hint are properly adjusted to SAML terms.

The tests need to check that the settings of the loginHint configuration in SAML identity provider respect the settings.